roles/lib_openshift/library/oc_route.py

```diff
@@ -1704,18 +1690,6 @@ def run_ansible(params, check_mode=False):
                          'content': params['key_content'],
                          'value': None, }, }
 
-        if params['tls_termination'] and params['tls_termination'].lower() != 'passthrough':  # E501
-
-            for key, option in files.items():
-                if key == 'destcacert' and params['tls_termination'] != 'reencrypt':
-                    continue
-
-                option['value'] = OCRoute.get_cert_data(option['path'], option['content'])  # E501
```

Comment:
On the other hand, there are 3 termination types: `edge`, `passthrough`, and `reencrypt`.
`passthrough` of course doesn't require any of the files.
When running oc_route on 3.6 with `reencrypt` or `edge`, and no keys or certificates the route object gets created. for `reencrypt` a default `destinationCACertificate` is created, and the route works. for `edge` no certificate is defined. I don't know if the route works, but it does get created. 
The question is what is the modules policy? does it prevent the user from defining objects that the API allows? If it doesn't, I think it should skip the validation and allow creation.